sdb/internal/cli.py

```diff
@@ -147,26 +157,88 @@ def load_debug_info(prog: drgn.Program, dpaths: List[str]) -> None:
             print("sdb: " + path + " is not a regular file or directory")
 
 
+def strings(fname: str, minlen: int = 4, maxlen: int = 16384) -> Iterable[str]:
+    """
+    Generate strings of at least minlen from the binary file.
+    """
+    with open(fname, errors="ignore") as f:
+        result = ""
+        for c in f.read(maxlen):
+            if c in string.printable:
+                result += c
+                continue
+            if len(result) >= minlen:
+                yield result
+            result = ""
+        if len(result) >= minlen:  # catch result at EOF
+            yield result
+
+
+def detect_from_core(core: str) -> str:
+    """
+    Parse through the core looking for OSRELEASE=<core>
+    """
+    match = "OSRELEASE"
+    line = ""
+    vm = ""
+    for s in strings(core, len(match), 16384):
+        if s.startswith(match):
+            line = s
+            break
+    if line:
+        split_list = line.split("\n")
+        if len(split_list) > 1:
+            line = split_list[0]
+        split_list = line.split("=")
+        if len(split_list) > 1:
+            vm = split_list[1]
+    return vm
+
+
+def is_core(filename: str) -> bool:
+    """
+    Parse beginning of the given file to see if it starts with 'KDUMP'
+    """
+    match = "KDUMP"
+    for s in strings(filename, len(match), 64):
+        if s.startswith(match):
+            return True
+    return False
```

Comment:
This is a bit hacky. How about we use the Python API of https://github.com/ptesarik/libkdumpfile to first check if it is a valid kdump file and then use query its build-id parameter/property to get our matching?

Reply:
Can you give me some hints on which API calls I could use?  I'm not able to find a clean way to do this with libkdumpfile.